src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs

```diff
@@ -92,6 +107,19 @@ public static class AspireAzureCosmosDBExtensions
             });
         }
 
+        // If the user opts into ignoring the emualtor certificate and the emulator is detected
+        // via the use of the hard coded account key, then disable certificate checking for the
+        // Cosmos DB client.
+        if (settings.IgnoreEmulatorCertificate && IsEmulatorConnectionString(settings.ConnectionString))
+        {
+            clientOptions.HttpClientFactory = () => new HttpClient(new HttpClientHandler()
+            {
+                ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator
+            });
+            clientOptions.ConnectionMode = ConnectionMode.Gateway;
```

Comment:
I think this is the best thing to do, but it might be worth documenting that this limits the connection to Gateway mode which means that (for example) tracing will look significantly different when using the emulator versus production if their production mode is `Direct`.

Reply:
Its the work around documented by the Cosmos team :)

Reply:
Oh, I know - just worth pointing out the side effects of the workaround too.